enumerables.rb

```diff
@@ -0,0 +1,286 @@
+              break
+            end
+          end
+        end
+      elsif cond.is_a?(Numeric) || cond.is_a?(String)
+        my_each do |e|
+          if cond == e
+            classflag = true
+            break
+          end
+        end
+      end
+    elsif block_given?
+      my_each do |e|
+        if yield(e)
+          classflag = true
+          break
+        end
+      end
+    elsif cond.nil? and !block_given?
+      classflag = if empty?
+              false
+            else
+              true
+            end
+    else
+      false
+    end
+    classflag
+  end
+
+  # my_none Method
+  def my_none?(cond = nil)
+    classflag = true
+    if !cond.nil?
+      if cond.is_a?(Class)
+        my_each do |e|
+          if e.is_a?(cond)
+            classflag = false
+            break
+          end
+        end
+      elsif cond == Numeric
+        my_each do |e|
+          if e.is_a?(Numeric)
+            classflag = false
+            break
+          end
+        end
+      elsif cond.is_a?(Boolean)
+        my_each do |e|
+          if e == cond
+            classflag = false
+            break
+          end
+        end
+      elsif cond != Class and cond.is_a?(Regexp)
+        my_each do |e|
+          if e.is_a?(String)
+            if e.match?(cond)
+              classflag = false
+              break
+            end
+          end
+        end
+      elsif cond.is_a?(Numeric) || cond.is_a?(String)
+        my_each do |e|
+          if cond == e
+            classflag = false
+            break
+          end
+        end
+      end
+
+    elsif block_given?
+      my_each do |e|
+        if yield(e)
+          classflag = false
+          break
+        end
+      end
+    elsif cond.nil? and !block_given?
+      classflag = if to_a.empty?
+        true
+      else
+        false
+      end
+    else
+      true
+    end
+    classflag
+  end
+
+  # my_none Method (another approach), didn't work because we couldn't use yield on a yield...
+  # def my_none_2?(cond = nil)
+  #   res = false
+  #   if !cond.nil?
+  #     inv = self.my_any?(cond)
+  #   elsif block_given?
+  #     block = yield
+  #     inv = self.my_any? {yield.to_s}
+  #   end
+  #   inv == true ? res = false : res = true
+  #   res
+  #   block
+  # end
+
+  # my_count Method
+  def my_count(cond = nil)
+    counter = 0
+    if block_given?
+      my_each do |f|
+        counter += if yield f
+        end
+        counter
+      end
+    elsif cond.nil?
+      length
+    elsif !cond.nil?
+      my_each do |e|
+        counter += 1 if cond == e
+      end
+      counter
+    end
+  end
+
+  # my_map method
+  def my_map(pro = nil)
+    new_array = []
+    if pro.is_a?(Proc)
+      my_each do |e|
+        new_array.push(pro.call(e))
+      end
+      new_array
+    elsif block_given?
+      my_each do |e|
+        new_array.push(yield e)
+      end
+      new_array
+    else
+      new_array = self
+    end
+    new_array
+  end
```

Comment:
- [ ] should have working `my_map`  identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))

#### my_map

- [ ] returns an Enumerator if no block is given

```ruby
[1, 2, 5].my_map == [1, 2, 5].map => #<Enumerator: [1, 2, 5]:map>
```
- my_map retrieves the wrong answer
![image](https://user-images.githubusercontent.com/52209067/95550463-2db5dd00-09c6-11eb-9b3d-e52e530c3034.png)

- Real map
![image](https://user-images.githubusercontent.com/52209067/95550490-39090880-09c6-11eb-8580-8c65702a6fed.png)